pkg/storage/tsdb/multi_file_index.go

```diff
@@ -92,80 +92,68 @@ func (i *MultiIndex) forIndices(ctx context.Context, from, through model.Time, f
 	return results, nil
 }
 
-func (i *MultiIndex) GetChunkRefs(ctx context.Context, userID string, from, through model.Time, shard *index.ShardAnnotation, matchers ...*labels.Matcher) ([]ChunkRef, error) {
+func (i *MultiIndex) GetChunkRefs(ctx context.Context, userID string, from, through model.Time, res *[]ChunkRef, shard *index.ShardAnnotation, matchers ...*labels.Matcher) error {
+	*res = (*res)[:0]
```

Comment:
Where is `(*MultiIndex).GetChunkRefs()` called from? If a newly retrieved `*[]ChunkRef` is passed to the function, why do we need to reset the slice?

Reply:
This isn't actually called from anywhere yet -- we're just building the prototype for how we will interact with the tsdb-based index. Regarding `*[]ChunkRef`, it was to allow the caller to supply their own slice to avoid allocations. It's encouraged that the slice could be reused across multiple calls to avoid allocations, so we reset it.